game-core/src/main/java/games/strategy/triplea/ai/proAI/data/ProTerritoryManager.java

```diff
@@ -115,51 +115,54 @@ public void populateEnemyDefenseOptions() {
 
     // Determine if territory can be successfully attacked with max possible attackers
     final List<Territory> territoriesToRemove = new ArrayList<>();
-    for (final Territory t : attackMap.keySet()) {
-      final ProTerritory patd = attackMap.get(t);
+    for (final Map.Entry<Territory, ProTerritory> territoryProTerritoryEntry : attackMap.entrySet()) {
+      final ProTerritory patd = territoryProTerritoryEntry.getValue();
 
       // Check if I can win without amphib units and ignore AA since max units might have lots of planes
       List<Unit> defenders = CollectionUtils.getMatches(patd.getMaxEnemyDefenders(player, data),
           ProMatches.unitIsEnemyAndNotAa(player, data));
       if (isIgnoringRelationships) {
-        defenders = new ArrayList<>(t.getUnits().getUnits());
+        defenders = new ArrayList<>((territoryProTerritoryEntry.getKey()).getUnits().getUnits());
```

Comment:
Uneccessary parentheses, and the second call to `getUnits()` is uneccessary, because UnitCollection implements Collection already.